datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.java

```diff
@@ -165,7 +166,11 @@ public void onLocationChanged(final Location location) {
 
             synchronized (this) {
                 for (CapturingProcessListener listener : this.listener) {
-                    listener.onLocationCaptured(new GeoLocation(latitude, longitude, gpsTime, speed, gpsAccuracy));
+                    try {
+                        listener.onLocationCaptured(new GeoLocation(latitude, longitude, gpsTime, speed, gpsAccuracy));
+                    } catch (DataCapturingException e) {
+                        throw new IllegalStateException(e);
```

Comment:
IllegalStateException is thrown when location data is captured without GpsFix (i.e. timestamp = 0L), that should be legit, right?

Reply:
It is legit for framework functions, which you cannot change to throw a DataCapturingException. For all other cases use a typed exception.